### PR TITLE
fix: four independently-verified bugs (#905, #908, #886, #909)

### PR DESCRIPTION
**Pull request by SomSamantray**
## Summary

Four small, independent bug fixes, each verified against current `main` before implementing (issue trackers can lag reality — several adjacent issues turned out to already be fixed, see note below):

1. **#905** — `CONFIGURATION.md` documents `--web-backend=keyless` to force the zero-key web-search floor, but argparse's `choices` list rejected it. The runtime (`grounding.py`, `pipeline.py`) already handled `"keyless"` correctly — only the CLI parser was wrong. Added it to `choices`.
2. **#908** — arXiv's exact-phrase quoted query returns zero results for any natural-language multi-word topic (a phrase like "AI video generation advances" almost never appears verbatim in a title/abstract). Now retries once, unquoted, only when the quoted query returns a genuinely clean empty response (no CLI error) — a real failure still skips the retry.
3. **#886** — Source URLs in the saved raw report ("All Items by Source") and the internal evidence-cluster output (`_render_candidate`) rendered as plain text instead of markdown links. Fixed both; also added a missing empty-URL guard to `_render_candidate` (it had none, unlike the item-loop location, which would have produced a broken `[]()` for an empty `candidate.url`).
4. **#909** — Truth Social requests get HTTP 403 from Cloudflare because of the skill's default User-Agent, regardless of token validity. Added the same browser-like headers (`User-Agent: http.BROWSER_USER_AGENT`, `Accept`, `Accept-Language`, `Referer`) already used for the keyless Reddit path — the same class of fix already applied to Reddit in #199.

## Known limitations (flagged during review, left as follow-up)

- **#908's retry roughly doubles arXiv latency for the common natural-language-topic case.** In Discovery/library-feed flows, many nominations share one enrichment deadline; doubling arXiv's per-topic time budget could push more nominations past that deadline into degraded (nomination-only) enrichment. A shorter retry timeout or a discover-context skip would address this, but that's a judgment call beyond this fix's scope — flagging for your consideration rather than guessing at the right trade-off.
- The unquoted retry has no escaping for arXiv query-syntax tokens (`AND`/`OR`/colons) — low-severity edge case, inherent to what "unquoted" search means.

## Testing

- `python -m pytest tests/test_cli_v3.py tests/test_arxiv_source.py tests/test_render_v3.py tests/test_truthsocial.py tests/test_registers.py -v` — all new and existing tests pass.
- Full suite run locally: confirmed the ~34 pre-existing failing tests (POSIX permission-bit assertions, `killpg`, unwritable-dir checks — native-Windows environment gaps unrelated to `uv`) are present identically with or without this diff.
- `test_registers.py`'s pinned byte-identity hash for the default render register was intentionally updated — #886 changes that exact rendered output (markdown links instead of plain URLs).
- Ran `ce-code-review` (correctness, project-standards, testing, adversarial personas): no P0/P1 findings; verdict "Ready to merge."

## Changelog

Added `changelog.d/905.fixed.md`, `changelog.d/908.fixed.md`, `changelog.d/886.fixed.md`, `changelog.d/909.fixed.md` per `CONTRIBUTING.md`.

## Agent disclosure

This PR was prepared with Claude Code, from repo analysis and bug verification through implementation and review. All commits, tests, and this description were reviewed before submission.

## Relationship

Independent of #911 (my other open PR, unrelated Windows npx fix). No known relationship to other open PRs on these four issues.

### CODE REVIEW

**Comment by tmchow** [comment]
@SomSamantray please do not create PRs like this in the future putting so many fixes into the same PR. It makes it hard to review and hard to unpack.  Agents are good at breaking things apart and you even say " independent bug fixes,"

File: skills/last30days/scripts/lib/render.py [hunk 1]
```diff
@@ -1561,6 +1561,44 @@ def render_comparison_multi_context(
...
+    ``html.escape(url, quote=True)``. Falls back to the bare (sanitized) URL
+    string rather than guessing at an escaping scheme for markdown syntax.
+    """
+    if not url:
+        return ""
+    url = _sanitize_url_for_single_line_output(url)
+    if not url.startswith(_SAFE_MARKDOWN_LINK_SCHEMES):
+        return url
+    if any(ch in url for ch in _MARKDOWN_LINK_UNSAFE_CHARS):
+        return url
```
[the remark this answers is not in the document]
Comment (tmchow) [reply skills/last30days/scripts/lib/render.py]: > When a source URL contains Markdown delimiters, `_markdown_url_link` returns the URL unchanged, so embedded syntax such as `[click](javascript:alert)` remains an active, deceptively labeled link in saved reports and internal evidence output.

Fixed in b3e87fd — rejected URLs now render as escaped, inert Markdown text after single-line sanitization; safe HTTP(S) URLs remain clickable at both PR-added output sites. Added regressions for embedded links, raw HTML/autolinks, delimiters, and both render paths.

**Comment by tmchow** [comment]
> This PR is not yet safe to merge because delimiter-bearing source URLs can still inject active Markdown links into saved reports and evidence output.

Addressed in b3e87fd: rejected URLs are now escaped as inert text, with regression coverage across both PR-added render paths. I also corrected the arXiv fallback query to explicit AND-conjoined field clauses and made malformed CLI responses fail instead of triggering a misleading retry.
